### PR TITLE
Multicluster checks

### PR DESCRIPTION
**Pull request by zaharidichev**
Adds multicluster checks that: 

- check that a service mirror controller exists (1 only)
- check that the expected ClusterRoles, ClusterRoleBindings and ServiceAccounts for the controller exist locally
- for each secret of type `remote-kubeconfig` we check that we can parse the contents into a k8s RestConfig object, can instantiate API connection to the remote cluster AND can LIST, WATCH and GET services in all namespace on this cluster. This is done for all clusters and if there is anything wrong a list of clusters violating these constraints is printed
- checks that each of the gateways (for which we have metrics in Prometheus) is alive and if not it prints a list of problematic cluster gateways

### CODE REVIEW

File: charts/linkerd2-service-mirror/templates/service-mirror.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ metadata:
   name: linkerd-service-mirror
   namespace: {{.Values.namespace}}
   labels:
-    {{.Values.controllerComponentLabel}}: linkerd-service-mirror
+    {{.Values.controllerComponentLabel}}: linkerd-service-mirrors
```
Comment (adleong) [inline charts/linkerd2-service-mirror/templates/service-mirror.yaml:35]: is this change intentional?

File: cli/cmd/check.go [hunk 2]
```diff
@@ -50,6 +52,7 @@ func (options *checkOptions) nonConfigFlagSet() *pflag.FlagSet {
 	flags.StringVarP(&options.namespace, "namespace", "n", options.namespace, "Namespace to use for --proxy checks (default: all namespaces)")
 	flags.BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
 	flags.BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
+	flags.BoolVar(&options.multicluster, "multicluster", options.multicluster, "Run multicluster checks")
```
Comment (adleong) [inline cli/cmd/check.go:55]: just thinking out loud here (figuratively):

would it make sense to automatically enable these checks if the service mirror component has been installed?  I think it's still useful to have the command so that if the service mirror component failed to be installed for some reason, a user running `linkerd check --multicluster` would be informed of that.  But I think it might also be useful to automatically enable this flag so that users who have successfully installed the service mirror will still get informed of errors with their multicluster set up, even if they don't know (or remember) to add the `--multicluster` flag.

File: controller/cmd/service-mirror/config_watcher.go [hunk 3]
```diff
@@ -112,74 +113,49 @@ func (rcw *RemoteClusterConfigWatcher) Stop() {
...
 		return fmt.Errorf("unable to parse kube config: %s", err)
 	}
 
 	rcw.Lock()
 	defer rcw.Unlock()
 
-	if _, ok := rcw.clusterWatchers[config.clusterName]; ok {
-		return fmt.Errorf("there is already a cluster with name %s being watcher. Please delete its config before attempting to register a new one", config.clusterName)
+	if _, ok := rcw.clusterWatchers[config.ClusterName]; ok {
+		return fmt.Errorf("there is already a cluster with name %s being watcher. Please delete its config before attempting to register a new one", config.ClusterName)
```
Comment (adleong) [inline controller/cmd/service-mirror/config_watcher.go:131]: typo: being watcher

File: pkg/healthcheck/healthcheck.go [hunk 4]
```diff
@@ -1463,6 +1546,169 @@ func FetchLinkerdConfigMap(k kubernetes.Interface, controlPlaneNamespace string)
...
+		return fmt.Errorf("Service mirror ClusterRole is missing expected policy rule")
+	}
+
+	if err := compare(expectedServiceMirrorPolicyVerbs, policyRule.Verbs); err != nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing verbs: %s", err)
+	}
+
+	if err := compare(expectedServiceMirrorPolicyResources, policyRule.Resources); err != nil {
+		return fmt.Errorf("Service mirror ClusterRole is missing required resources: %s", err)
+	}
```
Comment (alpeb) [inline pkg/healthcheck/healthcheck.go:1618]: Can you avoid short-circuiting on the first error here, and instead show both errors if present?

File: pkg/healthcheck/healthcheck.go [hunk 5]
```diff
@@ -1463,6 +1546,169 @@ func FetchLinkerdConfigMap(k kubernetes.Interface, controlPlaneNamespace string)
...
+		if err := hc.checkCanPerformAction(remoteAPI, "list", corev1.NamespaceAll, "", "v1", "services"); err == nil {
+			canList = true
+		}
+
+		if err := hc.checkCanPerformAction(remoteAPI, "watch", corev1.NamespaceAll, "", "v1", "services"); err == nil {
+			canWatch = true
+		}
+
+		if !canGet || !canList || !canWatch {
+			errors = append(errors, fmt.Sprintf("* cluster: [%s]: Insufficient permissions GET: %v, LIST: %v, WATCH: %v", config.ClusterName, canGet, canList, canWatch))
```
Comment (alpeb) [inline pkg/healthcheck/healthcheck.go:1673]: Nit: could you unify the error format with the one from `checkServiceMirrorLocalRBAC()`? (there you have something like `Service mirror ClusterRole is missing verbs: expected create,delete,get,list,update,watch, got create,delete,get,list,watch`

File: cli/cmd/check.go (hunk 2 above)
Comment (zaharidichev) [reply cli/cmd/check.go:55]: Yes.. that actually makes a lot of sense. Will include that modification

Comment (zaharidichev) [reply cli/cmd/check.go:55]: @adleong I pushed a commit that addresses that, can you take a look.
